language-plutus-core/src/Language/PlutusCore/TypeCheck/Internal.hs

```diff
@@ -287,18 +285,27 @@ typeOfBuiltinName
     => BuiltinName -> Type TyName uni ()
 typeOfBuiltinName bn = withTypedBuiltinName bn $ typeOfTypedBuiltinName @(Term TyName Name _ ())
 
--- | @unfoldFixOf pat arg k = NORM (vPat (\(a :: k) -> ifix vPat a) arg)@
-unfoldFixOf
+-- | @unfoldIFixOf pat arg k = NORM (vPat (\(a :: k) -> ifix vPat a) arg)@
+unfoldIFixOf
     :: Normalized (Type TyName uni ())  -- ^ @vPat@
     -> Normalized (Type TyName uni ())  -- ^ @vArg@
     -> Kind ()                          -- ^ @k@
     -> TypeCheckM uni ann (Normalized (Type TyName uni ()))
-unfoldFixOf pat arg k = do
+unfoldIFixOf pat arg k = do
     let vPat = unNormalized pat
         vArg = unNormalized arg
     a <- liftQuote $ freshTyName "a"
+    -- We need to rename @vPat@, otherwise it would be used twice below, which would break global
+    -- uniqueness. Alternatively, we could use 'normalizeType' instead of 'normalizeTypeM' as the
+    -- former performs renaming before doing normalization, but renaming the entire type implicitly
+    -- would be less efficient than renaming a subpart of the type explicitly.
+    --
+    -- Note however that breaking global uniqueness here most likely would not result in buggy
+    -- behavior, see https://github.com/input-output-hk/plutus/pull/2219#issuecomment-672815272
```

Comment:
Maybe just copy in the content?

Reply:
It's kinda long and it's more of a curious thing than something we're actually interested in, so I don't want to pollute implementation and its docs with largely irrelevant content. I did copy the content in `Language.PlutusCore.Examples.Data.Shad` (and added more), because there it's relevant, but I don't want to reference that file as we can move it somewhere else and such references outdate pretty quickly. A GitHub link will not outdate any time soon, so I think it's reasonable to add a link instead of a full explanation.